django/http/request.py

```diff
@@ -97,39 +97,47 @@ def _set_content_type_params(self, meta):
             else:
                 self.encoding = self.content_params['charset']
 
-    def _get_raw_host(self):
```

Comment:
I thought you wanted to keep `_get_raw_host`

Reply:
Nope not really. All I wanted was to have one function while keeping the behaviour mostly as is and without code duplication.